src/types.ts

```diff
@@ -63,35 +39,37 @@ export type ResourceRelationship = {
   id: string;
 };
 
-export type Metadata = {
-  [key: string]: string | number | boolean | Metadata;
+export type Meta = {
+  [key: string]: string | number | boolean | Meta;
 };
 
 export type JsonApiDocument<
   ResourceT = Resource,
   RelatedResourcesT = Resource
 > = {
   data: ResourceT | ResourceT[];
-  errors?: JsonApiError[];
-  meta?: Metadata;
+  meta?: Meta;
   included?: RelatedResourcesT[];
 };
 
-export type JsonApiErrorDocument<ErrorCodeT = ErrorCode> = {
-  errors?: JsonApiError<ErrorCodeT>[];
-  meta?: Metadata;
-};
-
-export type JsonApiRequest = {
-  data?: Resource;
-  meta?: Metadata;
+export type JsonApiErrorsDocument = {
```

Comment:
@Exelord this seems a bit counterintuitive in terms of the JSONAPI spec, which says that a document _may_ have one of four top-level elements: `data`, `meta`, `included`, `errors`. Either we bring the `errors[]` collection back to `JsonApiDocument` or we make it a mandatory field in `JsonApiErrorsDocument` (`errors` being optional in this type makes little sense).

Reply:
Do not forget these rules:
> The data member MUST NOT coexist with either the errors or operations members in the same document.

> The operations member MUST NOT coexist with the errors member in the same document.

So basically there is no chance data can coexist with errors.

Yes I agree, JsonApiErrorsDocument should require errors :)